### PR TITLE
Fc 968 basket save location

### PR DESCRIPTION
**Pull request by desderry**

### CODE REVIEW

File: src/Web/Sfa.Das.Sas.Shared.Components/ViewComponents/Basket/AddOrRemoveFromBasket/AddOrRemoveFromBasketViewComponent.cs [hunk 1]
```diff
@@ -19,11 +19,12 @@ public AddOrRemoveFromBasketViewComponent(IMediator mediator, ICookieManager coo
             _cookieManager = cookieManager;
         }
 
-        public async Task<IViewComponentResult> InvokeAsync(string apprenticeshipId, int? ukprn = null)
+        public async Task<IViewComponentResult> InvokeAsync(string apprenticeshipId, int? ukprn = null, int? location = null)
```
Comment (leewadhams) [inline src/Web/Sfa.Das.Sas.Shared.Components/ViewComponents/Basket/AddOrRemoveFromBasket/AddOrRemoveFromBasketViewComponent.cs]: Does the locationId not need to be passed in the `IsInBasket` method?

File: src/Web/Sfa.Das.Sas.Shared.Components/Controllers/BasketController.cs [hunk 2]
```diff
@@ -73,11 +73,10 @@ public async Task<IActionResult> AddProviderFromDetails(SaveBasketFromProviderDe
         [HttpPost]
         public async Task<IActionResult> AddProviderFromResults(SaveBasketFromProviderSearchViewModel queryModel)
         {
-            if (!await IsInBasket(queryModel.SearchQuery.ApprenticeshipId, queryModel.ItemId))
+            if (!await IsInBasket(queryModel.SearchQuery.ApprenticeshipId, queryModel.Ukprn,queryModel.LocationId))
```
Comment (leewadhams) [inline src/Web/Sfa.Das.Sas.Shared.Components/Controllers/BasketController.cs]: Does `AddProviderFromDetails` also need locationId adding in to it's `IsInBasket()` check?

File: src/Web/Sfa.Das.Sas.ApplicationServices/Handlers/AddorRemoveFavouriteInBasketCommandHandler.cs [hunk 3]
```diff
@@ -41,11 +41,24 @@ public async Task<Guid> Handle(AddOrRemoveFavouriteInBasketCommand request, Canc
...
+                    {
+                        basket.Remove(request.ApprenticeshipId, request.Ukprn.Value, request.Location.Value);
+                        basketChanged = true;
+                    }
+                    else
+                    {
+                        basketChanged = basket.Add(request.ApprenticeshipId, request.Ukprn.Value, request.Location.Value);
+                    }
+                }
+                else if (request.Ukprn.HasValue)
```
[the remark this answers is not in the document]
Comment (desderry) [reply src/Web/Sfa.Das.Sas.ApplicationServices/Handlers/AddorRemoveFavouriteInBasketCommandHandler.cs]: yep, for people who have already used the save favourites and they want to remove

File: src/Web/Sfa.Das.Sas.Shared.Components/Controllers/BasketController.cs (hunk 2 above)
Comment (desderry) [reply src/Web/Sfa.Das.Sas.Shared.Components/Controllers/BasketController.cs]: yep, i'll add it in. good spot!

**Pull request merged**